File: Week2/problem1.java
/*
Language: java
ID: 61067
QLink: https://quera.org/course/assignments/61067/problems
Author: @Amirkhaskar
*/

import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        long num = scanner.nextLong();
        System.out.println(fitsPattern(num));
    }

    static String fitsPattern(long num) {
        String numStr = Long.toString(num);
        int lenNum = numStr.length();

        if (lenNum % 2 == 0) {
            int half = lenNum / 2;
            int firstHalf = Integer.parseInt(numStr.substring(0, half));
            int secondHalf = Integer.parseInt(numStr.substring(half));

            long sumNumbers = 0;
            for (int i = firstHalf; i <= secondHalf; i++) {
                sumNumbers += i;
            }

            if (sumNumbers == num) {
                return "YES!";
            } else {
                return "NO!";
            }
        } else {
            return "NO!";
        }
    }
}

File: Week2/problem1.cs
/*
Language: csharp
ID: 61067
QLink: https://quera.org/course/assignments/61067/problems
Author: @Amirkhaskar
*/

using System;

class Program
{
    static string FitsPattern(int num)
    {
        string numStr = num.ToString();
        int lenNum = numStr.Length;

        if (lenNum % 2 == 0)
        {
            int half = lenNum / 2;
            int firstHalf = int.Parse(numStr.Substring(0, half));
            int secondHalf = int.Parse(numStr.Substring(half));

            int sumNumbers = 0;
            for (int i = firstHalf; i <= secondHalf; i++)
            {
                sumNumbers += i;
            }

            if (sumNumbers == num)
            {
                return "YES!";
            }
            else
            {
                return "NO!";
            }
        }
        else
        {
            return "NO!";
        }
    }

    static void Main()
    {
        int num = int.Parse(Console.ReadLine());
        Console.WriteLine(FitsPattern(num));
    }
}
